<?xml version="1.0"?>
<launch>

<!--Input######################################################################################################-->
    <param name="/dataset_pub_delay"      type="double"  value="3.0" />
    <param name="/dataset_pub_rate"       type="int"     value="10" />
    <param name="/publish_gt"                type="bool"    value="true" />
    <param name="/dataset_folder_path"    type="string"  value="$(find flvis)/bag/KITTI/dataset/sequences/00/" />
    <param name="/dataset_gt_file"        type="string"  value="$(find flvis)/bag/KITTI/dataset/poses/00.txt" />
    <node pkg="flvis" type="kitti_publisher" name="kitti_publisher" output="screen"/>

<!--FLVIS######################################################################################################-->
    <arg name="node_start_delay"  default="1.0" />
    <param name="/yamlconfigfile" type="string" value="$(find flvis)/launch/KITTI/KITTI.yaml"/>
    <param name="/voc"            type="string" value="$(find flvis)/voc/voc_orb.dbow3"/>

    <!-- Manager -->
    <node pkg="nodelet" type="nodelet"
          name="flvis_nodelet_manager" args="manager" output="screen"
          launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' ">
          <param name="num_worker_threads" value="4" />
    </node>

    <!-- TrackingNode -->
    <!-- D435i -->
    <node pkg="nodelet" type="nodelet" args="load flvis/TrackingNodeletClass flvis_nodelet_manager"
        name="TrackingNodeletClass_loader" output="screen"
        launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' ">
        <remap from="/imu"               to="/camera/imu"/>
    </node>

    <!-- LocalMapNode -->
    <!--window_size: Num of keyframes in sliding window optimizer-->
<!--    <node pkg="nodelet" type="nodelet" args="load flvis/LocalMapNodeletClass flvis_nodelet_manager"
          name="LocalMapNodeletClass_loader" output="screen"
          launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' ">
        <param name="/window_size" type="int" value="8" />
    </node>-->

    <!-- LoopClosingNode -->
<!--    <node pkg="nodelet" type="nodelet" args="load flvis/LoopClosingNodeletClass flvis_nodelet_manager"
        name="LoopClosingNodeletClass_loader" output="screen"
        launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' ">
    </node>-->


</launch>
